docs/common-toolchains.md

```diff
@@ -31,10 +31,29 @@ The intention is to revise/update the definitions of the common
 toolchains regularly (see [Update cycle for common toolchains][common_toolchains_update_cycle]), which again
 can be a joint effort that benefits many HPC sites.
 
-Currently, two different common toolchains are being maintained: `foss`
+Currently, two different families of common toolchains are being maintained: `foss`
 and `intel`; see below for more details, and also
 [Overview of common toolchains][common_toolchains_overview].
 
+To be more helpful in understanding the differences between these families, **here is a diagram** that explains what is added in
+each additional layer:
+
+``` mermaid
+
+graph LR
+  A[GCCCore] --> |binutils| B[GCC];
+  A --> |binutils| C[iccifort]
+  B --> |MPI libs| E[gompi];
+  C --> |MPI libs| F[iimpi];
+  B --> |Blas, FFTW| D[gfbf];
+  E --> |Blas, FFTW, Lapack libs| G[foss];
+  F --> |Blas, FFTW, Lapack libs| H[intel];
+```
```

Comment:
improved graph:

``` mermaid
graph LR
  A[GCCCore] --> |binutils| B[GCC];
  A --> |binutils| C[iccifort]
  B --> |OpenMPI| E[gompi];
  C --> |impi| F[iimpi];
  B --> |FlexiBLAS + FFTW| D[gfbf];
  D --> |OpenMPI| G[foss];
  E --> |FlexiBLAS + FFTW| G[foss];
  F --> |imkl| H[intel];
```

Reply:
* should have different graph for old/new toolchains
* iccifort should be intel-compilers in recent toolchains
* iimkl subtoolchain is missing in intel/2022b and newer